forge-ai/src/main/java/forge/ai/ability/LifeLoseAi.java

```diff
@@ -149,6 +157,10 @@ protected boolean checkApiLogic(Player ai, SpellAbility sa) {
             return true;
         }
 
+        if (sa.getPayCosts() != null && sa.getPayCosts().hasSpecificCostType(CostSacrifice.class)) {
```

Comment:
these parts seem like another thing that's waaaay too broad
makes it sacrifice stuff like _Bile Urchin_ first chance after casting

Reply:
I think we can try limiting this to activating it in a threatened card scenario for now.

Reply:
this could misplay because you might be checking the wrong sacrifice target
but that aside, this should already be checked by the `activateForCost` call above anyway?
what about the other cases?

Reply:
Yeah, I think this is superfluous and risky, I'll remove it. I believe you're right it's handled by the other checks, I tested a couple scenarios and they still work. Do you have any particular other misplay situations you'd like me to check out? I'll probably do a better job figuring out how to wire it up properly if I know a specific situation where things go south :)

Reply:
for LifeGain it might be ok because it uses custom `willPayCosts`
but for which case does this help? 🤔

Reply:
Hmm, not sure off the top of my head either :/

Reply:
Ugh, can't really think of anything that would need a similar routine in LifeGain, I think we keep things as is for now in this part?...